stan/math/opencl/copy.hpp

```diff
@@ -144,14 +163,17 @@ inline matrix_cl packed_copy(const std::vector<double>& src, int rows) {
   if (dst.size() == 0) {
     return dst;
   }
-  cl::CommandQueue queue = opencl_context.queue();
   try {
+    cl::CommandQueue queue = opencl_context.queue();
     matrix_cl packed(packed_size, 1);
-    queue.enqueueWriteBuffer(packed.buffer(), CL_TRUE, 0,
-                             sizeof(double) * packed_size, src.data());
-    stan::math::opencl_kernels::unpack(cl::NDRange(dst.rows(), dst.rows()),
-                                       dst.buffer(), packed.buffer(),
-                                       dst.rows(), dst.rows(), triangular_view);
+    cl::Event packed_event;
+    queue.enqueueWriteBuffer(packed.buffer(), CL_FALSE, 0,
+                             sizeof(double) * packed_size, src.data(), NULL,
+                             &packed_event);
+    packed.add_write_event(packed_event);
+    stan::math::opencl_kernels::unpack(cl::NDRange(dst.rows(), dst.rows()), dst,
+                                       packed, dst.rows(), dst.rows(),
+                                       triangular_view);
```

Comment:
Unclear?

Reply:
Why wouldn't we? If we assume we want to manage event queues and clear them whenever we have the opportunity, and we decided a good opportunity is whenever things are copied in or out since that sort of naturally marks the boundaries of GPU computation, and this is one form of copy, why wouldn't we clear the queues here too?

Reply:
Ahh, that's what I was unclear about. Yeah that makes sense I'll do that

Reply:
So for this one, `dst` is actually created in the function so it's only going to have the events from the `unpack` kernel call. Do we to stop to clear that? idt we need to?